### PR TITLE
Add failure message in datasource metadata

### PR DESCRIPTION
**Pull request by heemin32**
### Description
Add failure message in datasource metadata based on comments in previous PR. https://github.com/opensearch-project/geospatial/pull/257/files#r1177299458
 
### Issues Resolved
N/A
 
### Check List
- [X] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by navneet1v — COMMENTED** [review]
Why we are making this change is not clear to me. @heemin32 can you add some context in the PR why we are doing this change.

**Comment by heemin32** [comment]
> Why we are making this change is not clear to me. @heemin32 can you add some context in the PR why we are doing this change.

Updated in the description.

File: src/main/java/org/opensearch/geospatial/ip2geo/action/PutDatasourceTransportAction.java [hunk 1]
```diff
@@ -112,20 +110,20 @@ public void onFailure(final Exception e) {
     protected void createDatasource(final Datasource datasource) {
         if (DatasourceState.CREATING.equals(datasource.getState()) == false) {
             log.error("Invalid datasource state. Expecting {} but received {}", DatasourceState.CREATING, datasource.getState());
-            markDatasourceAsCreateFailed(datasource);
+            markDatasourceAsCreateFailed(datasource, "Datasource state is not in CREATING");
```
[the remark this answers is not in the document]
Comment (heemin32) [reply src/main/java/org/opensearch/geospatial/ip2geo/action/PutDatasourceTransportAction.java:113]: Have to find out why it is happening. It shouldn't happen but if it happens, don't know why as of now.

**Comment by heemin32** [comment]
@navneet1v Any other comment?

File: src/main/java/org/opensearch/geospatial/ip2geo/action/PutDatasourceTransportAction.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (navneet1v) [reply src/main/java/org/opensearch/geospatial/ip2geo/action/PutDatasourceTransportAction.java:124]: I don't think so this is helpful. We should do some deep-dive on this.

**Pull request closed without merging**